plugins/providers/bigquery/config_test.go

```diff
@@ -11,6 +12,88 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+func TestCredentials(t *testing.T) {
+	encryptor := new(mocks.Encryptor)
+	decryptor := new(mocks.Decryptor)
+
+	t.Run("encrypt", func(t *testing.T) {
+		t.Run("should return error if creds is nil", func(t *testing.T) {
+			var creds *bigquery.Credentials
+			expectedError := bigquery.ErrUnableToEncryptNilCredentials
+
+			actualError := creds.Encrypt(encryptor)
+
+			assert.EqualError(t, actualError, expectedError.Error())
+		})
+
+		t.Run("should return error if encryptor failed to encrypt the creds", func(t *testing.T) {
+			creds := bigquery.Credentials{}
+			expectedError := errors.New("encryptor error")
+			encryptor.On("Encrypt", mock.Anything).Return("", expectedError).Once()
+
+			actualError := creds.Encrypt(encryptor)
+
+			assert.EqualError(t, actualError, expectedError.Error())
+		})
+
+		t.Run("should return encrypted password inside Credentials on success", func(t *testing.T) {
+			service_account_key := base64.StdEncoding.EncodeToString([]byte("service-account-key-json"))
+			resourceName := "resource_name"
+			creds := bigquery.Credentials{
+				ServiceAccountKey: service_account_key,
+				ResourceName:      resourceName,
+			}
+
+			expectedEncryptedServiceAccountKey := "encrypted_service_account_key"
+			encryptor.On("Encrypt", service_account_key).Return(expectedEncryptedServiceAccountKey, nil).Once()
+
+			actualError := creds.Encrypt(encryptor)
+
+			assert.Nil(t, actualError)
+			assert.Equal(t, expectedEncryptedServiceAccountKey, creds.ServiceAccountKey)
```

Comment:
let's also assert mock expectations
```suggestion
			assert.Equal(t, expectedEncryptedServiceAccountKey, creds.ServiceAccountKey)
			encryptor.AssertExpectations(t)
```